external/fv3fit/fv3fit/train_microphysics.py

```diff
@@ -204,6 +208,93 @@ def __post_init__(self) -> None:
             self.cache = False
 
 
+def get_model_output_sensitivities(
+    model: tf.keras.Model, sample: Mapping[str, tf.Tensor],
+):
+    """
+    Generate sensitivity jacabion for each output relative to a mean
+    value for each input
+    """
+
+    avg_profiles = {
+        name: tf.reduce_mean(data, axis=0, keepdims=True)
+        for name, data in sample.items()
+    }
+
+    # normalize factors so sensitivities are comparable but still
+    # preserve level-relative magnitudes
+    normalize_factors = {}
+    for name, data in sample.items():
+        centered_by_level = data - tf.reduce_mean(data, axis=0)
+        factor = tf.math.sqrt(tf.reduce_mean(centered_by_level ** 2))
+        normalize_factors[name] = factor
+
+    input_data = {name: avg_profiles[name] for name in model.input_names}
+
+    with tf.GradientTape(persistent=True) as g:
+        g.watch(input_data)
+        outputs = model(input_data)
+
+    all_jacobians = {}
+    for out_name in model.output_names:
```

Comment:
this will remove the dependence on the `model.output_names` attribute and allow this to work with any tf function code and be more robust.

```suggestion
    for out_name in outputs:
```